R/getDominant.R

```diff
@@ -90,7 +91,13 @@ setMethod("getDominant", signature = c(x = "SummarizedExperiment"),
         if(!is.null(rank)){
             x <- agglomerateByRank(x, rank, ...)
             mat <- assay(x, assay.type)
-        } # Otherwise, if "rank" is NULL, abundances are stored without ranking
+        # or factor that is specified by user
+        } else if (!is.null(f)) {
```

Comment:
How about just using "rank" without "by" and "f"

-->


if( !is.null(rank) %% rank %in% taxonomyRanks(tse)){
aggglomerateByRank
} else if( !is.null(rank) ){<
# Catch f and by so that they are not fed to agglomerateByVariable
agglomerateBYVariable
}

Reply:
> # Catch f and by so that they are not fed to agglomerateByVariable

I didn't get this part

Reply:
did you mean?

```
if( !is.null(rank) %% rank %in% taxonomyRanks(tse)){
            x <- agglomerateByRank(x, rank, ...)
        } else if (!is.null(rank)) {
            x <- agglomerateByVariable(x, "rows", rank , ...)
        }
        else {
       ...
        }
```